---
float32 charge_percent
bool charging